src/main/java/com/snowflake/kafka/connector/records/SnowflakeMetadataConfig.java

```diff
@@ -12,64 +19,37 @@ public class SnowflakeMetadataConfig {
 
   /** initialize with default config */
   public SnowflakeMetadataConfig() {
-    this(new HashMap<String, String>());
+    this(new HashMap<>());
   }
 
   /**
-   * set flag to false only if metadata config is set to false in config
+   * Set flag to false only if metadata config is not set to "true" in config.
    *
    * @param config a String to String map of configs
    */
   public SnowflakeMetadataConfig(Map<String, String> config) {
-    // have those local variable to avoid assigning to final values multiple times
-    // these values are the default values of the configuration
-    boolean createtime = true;
-    boolean topic = true;
-    boolean offsetAndPartition = true;
-    boolean all = true;
-    if (config.containsKey(SnowflakeSinkConnectorConfig.SNOWFLAKE_METADATA_CREATETIME)
-        && !config
-            .get(SnowflakeSinkConnectorConfig.SNOWFLAKE_METADATA_CREATETIME)
-            .equals(SnowflakeSinkConnectorConfig.SNOWFLAKE_METADATA_DEFAULT)) {
-      createtime = false;
-    }
-    if (config.containsKey(SnowflakeSinkConnectorConfig.SNOWFLAKE_METADATA_TOPIC)
-        && !config
-            .get(SnowflakeSinkConnectorConfig.SNOWFLAKE_METADATA_TOPIC)
-            .equals(SnowflakeSinkConnectorConfig.SNOWFLAKE_METADATA_DEFAULT)) {
-      topic = false;
-    }
-    if (config.containsKey(SnowflakeSinkConnectorConfig.SNOWFLAKE_METADATA_OFFSET_AND_PARTITION)
-        && !config
-            .get(SnowflakeSinkConnectorConfig.SNOWFLAKE_METADATA_OFFSET_AND_PARTITION)
-            .equals(SnowflakeSinkConnectorConfig.SNOWFLAKE_METADATA_DEFAULT)) {
-      offsetAndPartition = false;
-    }
-    if (config.containsKey(SnowflakeSinkConnectorConfig.SNOWFLAKE_METADATA_ALL)
-        && !config
-            .get(SnowflakeSinkConnectorConfig.SNOWFLAKE_METADATA_ALL)
-            .equals(SnowflakeSinkConnectorConfig.SNOWFLAKE_METADATA_DEFAULT)) {
-      all = false;
-    }
+    createtimeFlag = getMetadataProperty(config, SNOWFLAKE_METADATA_CREATETIME);
+    topicFlag = getMetadataProperty(config, SNOWFLAKE_METADATA_TOPIC);
+    offsetAndPartitionFlag = getMetadataProperty(config, SNOWFLAKE_METADATA_OFFSET_AND_PARTITION);
+    allFlag = getMetadataProperty(config, SNOWFLAKE_METADATA_ALL);
+  }
+
+  private static boolean getMetadataProperty(Map<String, String> config, String property) {
+    String value =
+        Optional.ofNullable(config.get(property))
+            .orElse(SnowflakeSinkConnectorConfig.SNOWFLAKE_METADATA_DEFAULT);
 
-    createtimeFlag = createtime;
-    topicFlag = topic;
-    offsetAndPartitionFlag = offsetAndPartition;
-    allFlag = all;
+    // Cannot use Boolean.parseBoolean in order to ensure backward compatibility.
+    // The equality check must be case sensitive.
+    return "true".equals(value);
```

Comment:
can't we break that contract? i mean - using case insensitive comparison wouldn't break old customers but would be a a nice improvement for new installations.

Reply:
I doubt anyone will use uppercase "TRUE" value to disable something.
I'd like this pr to introduce no changes, so I can change it in the following one.

Reply:
Create a JIRA or add TODO to make sure this won't fall out of the cracks?

Reply:
[It's done already](https://github.com/snowflakedb/snowflake-kafka-connector/pull/833/files#diff-cf910e10942ebdab56b8d5ab0021e53cf6ee64f7105482fcdca09a0c2ce41797R45) in one of the PRs following.